components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/WebNotification.kt

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.webnotifications
+
+import android.net.Uri
+
+/**
+ * A notification sent by the Web Notifications API.
+ *
+ * @property origin The website that fired this notification.
+ * @property title Title of the notification to be displayed in the first row.
+ * @property body Body of the notification to be displayed in the second row.
+ * @property tag Tag used to identify the notification.
+ * @property imageUrl Large image to display in the notification.
+ * Corresponds to [android.app.Notification.BigPictureStyle.bigPicture].
+ * @property iconUrl Medium image to display in the notification.
+ * Corresponds to [android.app.Notification.Builder.setLargeIcon].
+ * @property badgeUrl Small badge to display in the corner of the notification.
+ * Corresponds to [android.app.Notification.Builder.setSmallIcon].
+ * @property vibrate Vibration pattern felt when the notification is displayed.
+ * @property timestamp Time when the notification was created.
+ * @property renotify Preference flag that indicates the end user should be alerted after the show
+ * steps have run with a new notification that has the same tag as an existing notification.
+ * @property silent Preference flag that indicates no sounds or vibrations should be made.
+ * @property actions Action buttons to display below the notification.
+ * @property onClick Callback called with the selected action, or null if the main body of the
+ * notification was clicked.
+ * @property onClose Callback called when the notification is dismissed.
+ */
+data class WebNotification(
```

Comment:
That probably will need to live in `concept-engine`? 🤔 

I wonder if it is a bit early to write this code here. The GV API doesn't even exist yet. Looking at the WIP patch for GV it doesn't have half of the fields here.

Reply:
The intention was to support all the fields in Android Components and let GeckoView incrementally build out support for everything here. I can wait for this to land until the GV side is ready, but I've been working with Alvina from the GV team on this patch.